webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPRuleRepository.scala

```diff
@@ -75,11 +75,14 @@ class RoLDAPRuleRepository(
    * Full((parent,directive)) : found the directive (directive.id == directiveId) in given parent
    * Failure => an error happened.
    */
-  def get(id:RuleId) : IOResult[Rule]  = {
+  def getOpt(id:RuleId) : IOResult[Option[Rule]]  = {
     ruleMutex.readLock(for {
       con     <- ldap
-      crEntry <- con.get(rudderDit.RULES.configRuleDN(id.value)).notOptional(s"Rule with id '${id.value}' was not found")
-      rule    <- mapper.entry2Rule(crEntry).toIO.chainError("Error when transforming LDAP entry into a rule for id %s. Entry: %s".format(id, crEntry))
+      crEntry <- con.get(rudderDit.RULES.configRuleDN(id.value))
+      rule    <- crEntry match {
+                   case None    => None.succeed
```

Comment:
this changes the semantic, and should at least be updated in the comment
(but I find this much easier to understand)

Reply:
I could let the `def get = getOpt.notOptionnal(...)`